### PR TITLE
[BugFix]: Fixed add_noise in LMSDiscreteScheduler

### PR DESCRIPTION
**Pull request by nicolas-dufour**
The formulation of add noise is LMSDiscreteScheduler is wrong  (see crowsonkb/k-diffusion#4). This PR fixes that

### CODE REVIEW

**Review by patil-suraj — APPROVED** [review]
Ah, good catch!

**Review by anton-l — CHANGES_REQUESTED** [review]
Hi @nicolas-dufour! For this particular sampler, the model needs to be trained similarly to DDPM/DDIM (i.e. with the current `add_noise`). That's what works for Stable Diffusion, as far as I know (let me know if I misunderstand something!). 
But if you find a way to convert the beta schedule closer to the original k-lms, feel free to update the PR :)

**Comment by nicolas-dufour** [comment]
Hum ok i was under the impression that add_noise was the method used to add noise to a latent since it's used in the example for img2img. I have tried to use the current add_noise with stable diffusion but it did not work and the only way i managed to make it work was to change add_noise to the proposed change

**Comment by patrickvonplaten** [comment]
> Hi @anton-l, I have tried to use the current add_noise with stable diffusion doing some img2img but it did not work and the only way i managed to make it work was to change add_noise to the proposed change.
> 
> From what i can see from k-diffusion they also use this noise technique at train time (https://github.com/crowsonkb/k-diffusion/blob/3f93c28088890e4d6bc593072739e1d6e759b392/k_diffusion/layers.py#L28). It's also what is proposed by the original Karras et al paper on which k-diffusion scheduler is based (https://arxiv.org/pdf/2206.00364.pdf Algo 2).

Thanks for the links here @nicolas-dufour - from reading the code a bit. I think the sampler can work with both cases, *e.g.* it can work for inference on a model that was trained on classic DDPM (as it's now) if we adding this line: https://github.com/huggingface/diffusers/blob/16172c1c7ef6ec721bfe4d0787313519157749a1/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L141 to the pipeline, but as @nicolas-dufour pointed out one should also be able to train the model with the original formula. 

Could we maybe allow for both and add a config parameter? Something like: `predict_ddpm_criterion=False/True` ? What do you think @anton-l ?

**Comment by patil-suraj** [comment]
>Could we maybe allow for both and add a config parameter? Something like: predict_ddpm_criterion=False/True

I'm in favor of this. As @nicolas-dufour said, the current `add_noise` doesn't work for imag2image, in-painting. So would be important to have this.

File: src/diffusers/schedulers/scheduling_lms_discrete.py [hunk 1]
```diff
@@ -124,10 +124,7 @@ def step(
         return {"prev_sample": prev_sample}
 
     def add_noise(self, original_samples, noise, timesteps):
-        alpha_prod = self.alphas_cumprod[timesteps]
-        alpha_prod = self.match_shape(alpha_prod, original_samples)
-
-        noisy_samples = (alpha_prod**0.5) * original_samples + ((1 - alpha_prod) ** 0.5) * noise
+        noisy_samples = original_samples + noise * self.sigmas[timesteps]
```
Comment (anton-l) [inline src/diffusers/schedulers/scheduling_lms_discrete.py:127]: @nicolas-dufour could you add a quick snippet of how you use the `add_noise` function in your experiments? Because at the moment it fails for batched inputs, and I would expect it to be implemented like so: 

```suggestion
        sigmas = self.match_shape(self.sigmas, noise)
        noisy_samples = original_samples + noise * sigmas[timesteps]
```

Comment (nicolas-dufour) [reply src/diffusers/schedulers/scheduling_lms_discrete.py:127]: Hey, I'm doing the following:
I have ```init_latents``` of batch_size 2. and then
```python
init_timestep = int(num_inference_steps * strength)+ offset
init_timestep = min(init_timestep, num_inference_steps)
timesteps = self.scheduler.timesteps[-init_timestep]
t_start = max(num_inference_steps - init_timestep + offset, 0)

timesteps = torch.tensor([timesteps] * batch_size, dtype=torch.long, device=self.device)
noise = torch.randn(init_latents.shape, generator=generator, device=self.device)
init_latents = init_latents+noise* self.scheduler.sigmas[t_start]
```

**Comment by patrickvonplaten** [comment]
@anton-l let's merge this and then move onto your follow-up PR?

**Review by anton-l — APPROVED** [review]
@nicolas-dufour thank you for the fix! Your investigation spawned another PR for img2img and K-LMS compatibility in general, so feel free to join the conversation in if you'd like: https://github.com/huggingface/diffusers/pull/270

**Comment by patrickvonplaten** [comment]
Awesome addition @nicolas-dufour :heart:
